addons/account/models/account_move.py

```diff
@@ -2327,6 +2327,12 @@ def write(self, vals):
                 move._check_fiscalyear_lock_date()
                 move.line_ids._check_tax_lock_date()
 
+            # Disallow modifying readonly fields on a posted move
+            readonly_fields = ('invoice_line_ids', 'line_ids', 'invoice_date', 'date', 'partner_id', 'partner_bank_id',
+                               'invoice_payment_term_id', 'currency_id', 'fiscal_position_id', 'auto_post', 'invoice_cash_rounding_id')
+            if move.state == "posted" and any(val in readonly_fields for val in vals):
+                raise UserError(_("This move has been posted. The field you're trying to change is now readonly."))
```

Comment:
need to add that to the .pot file

edit: actually, we don't care if we target master